FiveCalls/FiveCalls/IssuesViewModel.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  IssuesViewModel.swift
+//  FiveCalls
+//
+//  Created by Indrajit on 17/10/17.
+//  Copyright © 2017 5calls. All rights reserved.
+//
+
+import Foundation
+
+struct IssuesViewModel {
+    private let categories:[Category]
+    private let showAll: Bool
+    private let activeIssues: [Issue]
+
+    init(categories:[Category], showAll: Bool) {
+        self.categories = categories
+        self.showAll = showAll
+        var activeIssues: [Issue] = []
+        if !showAll {
+            categories.forEach { (category) in
+                activeIssues.append(contentsOf: category.issues.filter({ $0.inactive == false }))
+            }
+        }
+        self.activeIssues = activeIssues
+    }
+
+    func numberOfSections() -> Int {
+        if showAll {
+            return categories.count
+        }
+        return 1
+    }
+
+    func numberOfRowsInSection(section: Int) -> Int {
+        if showAll {
```

Comment:
This entire class splits functionality based on this flag. This is calling to be a different type :)

At this point I think it makes sense to have 2 view models:

- `ActiveIssuesViewModel`  and `CategorizedIssuesViewModel` (not married to that name)

Then we can push the appropriate logic into the right view model type.

Reply:
We'll also need to extract out a protocol for these 2 so the view controller can treat them equally.

Reply:
Makes sense. This will make the code much more cleaner - will try to get this in by today evening.